ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/operations/TStoreOperationTest.java

```diff
@@ -188,4 +189,32 @@ public void noGasRefundFromTransientState() {
 
     assertThat(result.getGasCost()).isEqualTo(OptionalLong.of(100L));
   }
+
+  @Ignore
+  @Test
+  public void transientStoragePerformanceTest() {
+    long initialGas = 30_000_000L;
+    long remainingGas = initialGas;
+    long numOfOps =
+        (long) (initialGas * .95)
+            / (gasCalculator.getTransientStoreOperationGasCost()
+                + gasCalculator.getTransientLoadOperationGasCost()
+                + gasCalculator.getVeryLowTierGasCost() * 4);
+    final TStoreOperation tstore = new TStoreOperation(gasCalculator);
+    final TLoadOperation tload = new TLoadOperation(gasCalculator);
+    final MessageFrame frame =
+        createMessageFrame(Address.fromHexString("0x18675309"), initialGas, remainingGas);
+
+    long startTime = System.nanoTime();
+    for (int i = 0; i < numOfOps; i++) {
+      frame.pushStackItem(UInt256.ONE);
+      frame.pushStackItem(UInt256.fromHexString("0x01"));
+      tstore.execute(frame, null);
+      frame.pushStackItem(UInt256.fromHexString("0x01"));
+      tload.execute(frame, null);
+      frame.popStackItem();
+    }
+    long endTime = System.nanoTime();
+    assertThat((endTime - startTime) / 1e6).isLessThan(5000);
```

Comment:
what's the actual result here? iiuc, it's just doing store 1 in 1, load 1, pop, which should be executable in less than 1s ideally

Reply:
In general not a good idea to test time in this type of test. Our CI is running on busy hardware. We have benchmarks elsewhere.

Reply:
> what's the actual result here? iiuc, it's just doing store 1 in 1, load 1, pop, which should be executable in less than 1s ideally

It's performing this `numOfOps` times (134,433 based on a `blockgaslimit` of 30M) and ensuring it runs less than 5 secs. It's highly hardware dependent which is why I marked it with `@Ignore` to be ran manually. On my laptop, it runs in < 300 ms.

@atoulme I see precompile benchmark tests. Are these the tests that you're referring to?

Reply:
yeah sorry i meant the whole thing executable in less than 1s, i was just concerned about the 5s limit being close to the result on your machine :)

Reply:
Yes, they'd be under the jmh tests.

Reply:
@atoulme I've removed this test and created a new one in the jmh suite. See `TransientStorageOperationBenchmark.java`.